Library/Homebrew/dev-cmd/boneyard-formula-pr.rb

```diff
@@ -0,0 +1,125 @@
+
+    formula_path = formula.path
+    bottle_block = File.read(formula_path).include? "  bottle do"
+    if ARGV.dry_run?
+      puts "brew update"
+      puts "brew tap homebrew/boneyard"
+      puts "cd #{formula.tap.path}"
+      cd formula.tap.path
+      puts "cp #{formula_path.relative_path_from(formula.tap.path)} #{Tap.fetch("homebrew", "boneyard").formula_dir}"
+      puts "git rm #{formula_path.relative_path_from(formula.tap.path)}"
+      unless File.exist? formula.tap.path/"tap_migrations.json"
+        puts "Creating tap_migrations.json for #{formula.tap.name}"
+        puts "git add #{formula.tap.path}/tap_migrations.json"
+      end
+      puts "Loading tap_migrations.json"
+      puts "Adding #{formula.name} to tap_migrations.json"
+    else
+      safe_system "brew", "update"
+      safe_system "brew", "tap", "homebrew/boneyard"
+      cd formula.tap.path
+      cp formula_path.relative_path_from(formula.tap.path), Tap.fetch("homebrew", "boneyard").formula_dir
+      safe_system "git", "rm", formula_path.relative_path_from(formula.tap.path)
+      unless File.exist? formula.tap.path/"tap_migrations.json"
+        (formula.tap.path/"tap_migrations.json").write <<-EOS.undent
+          {
+          }
+        EOS
+        safe_system "git", "add", formula.tap.path/"tap_migrations.json"
+      end
+      tap_migrations = Utils::JSON.load(File.read(formula.tap.path/"tap_migrations.json"))
+      tap_migrations[formula.name] = "homebrew/boneyard"
+      tap_migrations = tap_migrations.sort.inject({}) { |a, e| a.merge!(e[0] => e[1]) }
+      (formula.tap.path/"tap_migrations.json").atomic_write(JSON.pretty_generate(tap_migrations) + "\n")
+    end
+    unless Formula["hub"].any_version_installed?
+      if ARGV.dry_run?
+        puts "brew install hub"
+      else
+        safe_system "brew", "install", "hub"
+      end
+    end
+    branch = "#{formula.name}-boneyard"
+    puts "cd #{formula.tap.formula_dir}" if ARGV.dry_run?
+    cd formula.tap.formula_dir
+    if ARGV.dry_run?
+      puts "git checkout -b #{branch} origin/master"
+      puts "git commit --no-edit --verbose --message=\"#{formula.name}: boneyard\" -- #{formula_path} #{formula.tap.path}/tap_migrations.json"
+      puts "hub fork --no-remote"
+      puts "hub fork"
+      puts "hub fork (to read $HUB_REMOTE)"
+      puts "git push $HUB_REMOTE #{branch}:#{branch}"
+      puts "hub pull-request -m $'#{formula.name}: boneyard\\n\\nCreated with `brew boneyard-formula-pr`.'"
+    else
+      safe_system "git", "checkout", "-b", branch, "origin/master"
+      safe_system "git", "commit", "--no-edit", "--verbose",
+        "--message=#{formula.name}: boneyard",
+        "--", formula_path, formula.tap.path/"tap_migrations.json"
+      safe_system "hub", "fork", "--no-remote"
+      quiet_system "hub", "fork"
+      remote = Utils.popen_read("hub fork 2>&1")[/fatal: remote (.+) already exists./, 1]
+      odie "cannot get remote from 'hub'!" if remote.to_s.empty?
```

Comment:
You can simplify this to `unless remote` or `if remote.nil?`. If there is a match, it is guaranteed to be non-empty thanks to the `+` in the regular expression. If there is no match, `remote` will always be `nil`.